llm/src/text_data.py

```diff
@@ -156,15 +163,20 @@ def __len__(self) -> Optional[int]:
 
 def build_text_dataloader(cfg: DictConfig, device_batch_size: int):
     dataset = StreamingTextDataset(
-        split=cfg.dataset.split,
-        remote=cfg.dataset.remote,
         local=cfg.dataset.local,
+        remote=cfg.dataset.remote,
+        split=cfg.dataset.split,
         shuffle=cfg.dataset.shuffle,
-        prefetch=cfg.dataset.prefetch,
-        tokenizer_name=cfg.dataset.tokenizer_name,
-        max_seq_len=cfg.dataset.max_seq_len,
-        group_method=cfg.dataset.group_method,
+        predownload=cfg.dataset.get('predownload', 100_000),
+        tokenizer_name=cfg.dataset.get('tokenizer_name', 'gpt2'),
+        max_seq_len=cfg.dataset.get('max_seq_len', 2048),
+        group_method=cfg.dataset.get('group_method', 'truncate'),
```

Comment:
I feel like `max_seq_len` and `group_method ` shouldn't be set using a default...
we call this the Danger Zone

Reply:
Ok, let me see what it looks like to move a couple of these args up to be non-defaulted. I think it just makes the arg order a little less pretty, but it makes sense. I feel like `tokenizer_name` also falls in the same category.